zrml/orderbook/src/lib.rs

```diff
@@ -401,17 +394,22 @@ mod pallet {
             // to repatriate successfully, e.g. taker gets a little bit less
             // it should always ensure that the maker's request (maker_fill) is fully filled
             let taker_fill = Self::get_taker_fill(&order_data, maker_fill)?;
+            let order_account = Self::order_account(order_id);
+
+            if !T::AssetManager::reserved_balance(maker_asset, &maker).is_zero() {
+                T::AssetManager::repatriate_reserved_named(
+                    &Self::reserve_id(),
+                    maker_asset,
+                    &maker,
+                    &taker,
+                    taker_fill,
+                    BalanceStatus::Free,
+                )?;
```

Comment:
Why not `reserved_balance_named` here instead? In that way we wouldn't have conflicts with potential other pallets.

Additionally I think the asset could be an outcome asset, which includes the market id, then it's fine, but it could also be the base asset. Other problem we have here then is, that if you have another order on the order book with the same base asset, you would go into this branch, right?

One way to solve this is by putting a more fine-grained reserve identifier onto the reserve, which directly identifies the asset additionally to the pallet identifier.

Reply:
Both cases (named and same base asset) don't really matter. The only thing this lines of code are trying to achieve is to check if a asset is reservable in general or not and then execute the correct branch. This builds up on the pre-condition that during order creation, an asset that is reservable is always reserved and an asset that is not reservable is always transferred to the order account.

Assume we use non named reserves. If there is reserved balance, the asset is reservable. Thus `repatriate_reserve_named` will be called which fails if there is no named reserve for that pallet, account and asset.
Assume we create two orders with the same reservable base asset. During order creation the base asset was reserved. When someone wants to fill the order or when the owner wants to remove the order, the order is fetched from the DB and the maker is extracted. If there is reserved balance for that maker, it is unreserved. The reserve of the other order is untouched, assuming logic works properly. I think proper tests for those scenarios are sufficient.

Name reserved balance check: https://github.com/zeitgeistpm/zeitgeist/pull/1283/commits/5185e92e6585f2d332dc42f542de2080ffc7e1d4